src/plugins/discover/public/application/apps/main/components/sidebar/discover_field.tsx

```diff
@@ -99,205 +265,43 @@ export function DiscoverField({
-            defaultMessage: 'Remove field from table',
-          })}
-        >
-          <EuiButtonIcon
-            color="danger"
-            iconType="cross"
-            className={actionBtnClassName}
-            onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-              if (ev.type === 'click') {
-                ev.currentTarget.focus();
-              }
-              ev.preventDefault();
-              ev.stopPropagation();
-              toggleDisplay(f, isSelected);
-            }}
-            data-test-subj={`fieldToggle-${f.name}`}
-            aria-label={removeLabelAria}
-          />
-        </EuiToolTip>
-      );
-    }
-  };
-
-  const actionButton = getActionButton(field, selected);
+  }, [infoIsOpen]);
 
   if (field.type === '_source') {
     return (
       <FieldButton
         size="s"
         className="dscSidebarItem"
         dataTestSubj={`field-${field.name}-showDetails`}
-        fieldIcon={dscFieldIcon}
-        fieldAction={actionButton}
-        fieldName={fieldName}
+        fieldIcon={<DiscoverFieldTypeIcon field={field} />}
+        fieldAction={
+          <ActionButton
+            field={field}
+            isSelected={selected}
+            alwaysShow={alwaysShowActionButton}
+            toggleDisplay={toggleDisplay}
+          />
+        }
+        fieldName={<FieldName field={field} />}
       />
     );
   }
 
-  const getFieldInfoIcon = () => {
-    if (field.type !== 'conflict') {
-      return null;
-    }
-    return (
-      <EuiToolTip
-        position="bottom"
-        content={i18n.translate('discover.field.mappingConflict', {
-          defaultMessage:
-            'This field is defined as several types (string, integer, etc) across the indices that match this pattern.' +
-            'You may still be able to use this conflicting field, but it will be unavailable for functions that require Kibana to know their type. Correcting this issue will require reindexing your data.',
-        })}
-      >
-        <EuiIcon
-          tabIndex={0}
-          type="alert"
-          title={i18n.translate('discover.field.mappingConflict.title', {
-            defaultMessage: 'Mapping Conflict',
-          })}
-          size="s"
-        />
-      </EuiToolTip>
-    );
-  };
-
-  const fieldInfoIcon = getFieldInfoIcon();
-
-  const shouldRenderMultiFields = !!multiFields;
-  const renderMultiFields = () => {
-    if (!multiFields) {
-      return null;
-    }
-    return (
-      <React.Fragment>
-        <EuiTitle size="xxxs">
-          <h5>
-            {i18n.translate('discover.fieldChooser.discoverField.multiFields', {
-              defaultMessage: 'Multi fields',
-            })}
-          </h5>
-        </EuiTitle>
-        {multiFields.map((entry) => (
-          <FieldButton
-            size="s"
-            className="dscSidebarItem dscSidebarItem--multi"
-            isActive={false}
-            onClick={() => {}}
```

Comment:
:information_source: This empty callback caused the element to actually render as a button that didn't do anything. I removed that to fix that accessibility bug.